gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegistrationUnit.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.hive;
+
+import java.util.List;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import gobblin.annotation.Alpha;
+import gobblin.configuration.State;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+
+/**
+ * A class that represents a Hive table or partition.
+ *
+ * @author ziliu
+ */
+@Getter
+@Alpha
+public class HiveRegistrationUnit {
```

Comment:
1. I'd rather not depend on classes in hive-metastore if we are not using the metastore based approach, since there is a possibility that they may have backward incompatible changes.
2. There's a lot of commonality of Table and Partition, so it is convenient to make them have a common parent (`HiveRegistrationUnit`)